internal/error.go

```diff
@@ -459,6 +471,17 @@ func NewContinueAsNewError(ctx Context, wfn interface{}, args ...interface{}) er
 	return i.NewContinueAsNewError(ctx, wfn, args...)
 }
 
+// NewContinueAsNewErrorWithOptions creates ContinueAsNewError instance with additional options.
+func NewContinueAsNewErrorWithOptions(ctx Context, wfn interface{}, options ContinueAsNewErrorOptions, args ...interface{}) error {
```

Comment:
```suggestion
func NewContinueAsNewErrorWithOptions(ctx Context, options ContinueAsNewErrorOptions, wfn interface{}, args ...interface{}) *ContinueAsNewError {
```

To match things that accept options and varargs (e.g. `Client.ExecuteWorkflow`) let's put the options before the workflow. Also, I wonder if we would like to return the actual continue as new error so they can alter things if they'd like. @Quinn-With-Two-Ns - preference?

Reply:
> the options as the last parameter:

But it isn't last, it's before `args`. The concern was "accept options _and varargs_". Would definitely support options last if it could be last, but in this case, I think it's worth the workflow and args being next to each other instead of split. Would like @Quinn-With-Two-Ns's opinion here though.

> In addition, none of the constructor within this file returns the concrete type (such as *ContinueAsNewError, *ApplicationError.

Yeah, I don't have a strong opinion here. I am ok continuing to return just `error`. Can get others' opinions too.